support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -0,0 +1,268 @@
+import { css } from '@emotion/react';
+import {
+	from,
+	palette,
+	space,
+	textSans,
+	until,
+} from '@guardian/source-foundations';
+import { Column, Columns, Container } from '@guardian/source-react-components';
+import {
+	FooterLinks,
+	FooterWithContents,
+} from '@guardian/source-react-components-development-kitchen';
+import { useEffect, useState } from 'react';
+import {
+	number,
+	object,
+	type Output,
+	parse,
+	picklist,
+	record,
+	string,
+} from 'valibot'; // 1.54 kB
+import { Box, BoxContents } from 'components/checkoutBox/checkoutBox';
+import { CheckoutHeading } from 'components/checkoutHeading/checkoutHeading';
+import { Header } from 'components/headers/simpleHeader/simpleHeader';
+import { ContributionsOrderSummary } from 'components/orderSummary/contributionsOrderSummary';
+import { PageScaffold } from 'components/page/pageScaffold';
+import { SecureTransactionIndicator } from 'components/secureTransactionIndicator/secureTransactionIndicator';
+import type { Currency } from 'helpers/internationalisation/currency';
+import { currencies } from 'helpers/internationalisation/currency';
+import { renderPage } from 'helpers/rendering/render';
+import { trackComponentClick } from 'helpers/tracking/behaviour';
+import { GuardianTsAndCs } from 'pages/supporter-plus-landing/components/guardianTsAndCs';
+
+const darkBackgroundContainerMobile = css`
+	background-color: ${palette.neutral[97]};
+	${until.tablet} {
+		background-color: ${palette.brand[400]};
+		border-bottom: 1px solid ${palette.brand[600]};
+	}
+`;
+
+const columns = css`
+	position: relative;
+	color: ${palette.neutral[7]};
+	${textSans.medium()};
+	padding-top: ${space[2]}px;
+`;
+
+const secureTransactionIndicator = css`
+	margin-bottom: ${space[3]}px;
+	${from.tablet} {
+		margin-bottom: ${space[4]}px;
+	}
+`;
+
+const shorterBoxMargin = css`
+	:not(:last-child) {
+		${until.tablet} {
+			margin-bottom: ${space[2]}px;
+		}
+	}
+`;
+
+const searchParams = new URLSearchParams(window.location.search);
+const query = {
+	productId: searchParams.get('product'),
+	ratePlanId: searchParams.get('ratePlan'),
+	chargeId: searchParams.get('charge'),
+};
```

Comment:
I am trying to keep all routing and other rendering logic out of the play lib as the split often causes a large amount of context switching and cognitive load on having to try and remember where things are.

This might not scale too well - but I do think our application is simple enough to not have to scale really.

The URL for the screenshot would look like:
`https://support.thegulocal.com/uk/checkout?product=GuardianWeeklyDomestic&ratePlan=OneYearGift&charge=Subscription`

Reply:
Notice `charge` is removed is (for example 'Subscription'), this is used to differentiate between `subscribe/print` and `contribute/digital`?

Reply:
The plan is that a person buys a `ratePlan`. The charges are essentially internal and don't need to be exposed to the user.